File: spring_server/src/main/java/com/semothon/spring_server/crawling/entity/CrawlingInterest.java
package com.semothon.spring_server.crawling.entity;

import com.semothon.spring_server.interest.entity.Interest;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@ToString(of = {"crawlingInterestId"})
@Table(name = "crawling_interests",
        indexes = {
        },
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"crawling_id", "interest_id"})
        }
)
public class CrawlingInterest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long crawlingInterestId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "crawling_id", nullable = false)
    private Crawling crawling;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "interest_id", nullable = false)
    private Interest interest;

    public void updateCrawling(Crawling crawling){
        this.crawling = crawling;
        crawling.addCrawlingInterest(this);
    }

    public void updateInterest(Interest interest){
        this.interest = interest;
        interest.addCrawlingInterest(this);
    }
}

File: fastapi_server/app/models/crawling_interest.py
from sqlalchemy import Column, ForeignKey, Integer, UniqueConstraint
from sqlalchemy.orm import relationship
from database import Base

class CrawlingInterest(Base):
    __tablename__ = "crawling_interests"
    __table_args__ = (
        UniqueConstraint("crawling_id", "interest_id", name="uq_crawling_interest"),
    )

    crawling_interest_id = Column(Integer, primary_key=True, autoincrement=True)
    crawling_id = Column(Integer, ForeignKey("crawling.crawling_id"), nullable=False)
    interest_id = Column(Integer, ForeignKey("interests.interest_id"), nullable=False)

    crawling = relationship("Crawling")
    interest = relationship("Interest")